### PR TITLE
Add initialCollapsed prop to LegendWidget

### PR DESCRIPTION
**Pull request by Clebal**
image

Changes made:

- Allow to render LegendWidget initially collapsed.
- Added TS types, prop types and JSDocs.

Breaking change:
- LegendWidgetUI is now dummy.

### CODE REVIEW

File: packages/react-widgets/src/widgets/LegendWidget.js [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
+ * Renders a <LegendWidget /> component
+ * @param  {object} props
+ * @param  {string} props.className - CSS class name
+ * @param  {string} props.initialCollapsed - Define initial collapsed value. false by default.
+ */
+function LegendWidget({ className, initialCollapsed }) {
   const dispatch = useDispatch();
   const layers = useSelector((state) =>
-    Object.values(state.carto.layers).filter((layer) => !!layer.legend)
+    [...Object.values(state.carto.layers), ...Object.values(state.carto.layers)].filter(
```
Comment (Josmorsot) [inline packages/react-widgets/src/widgets/LegendWidget.js:16]: Could you explain what are you doing here? Maybe you can create a selector to be more specific

**Review by Josmorsot — APPROVED** [review]
LGTM, just a minor question/suggestion

File: packages/react-widgets/src/widgets/LegendWidget.js (hunk 1 above)
Comment (Clebal) [reply packages/react-widgets/src/widgets/LegendWidget.js:16]: Ups, sorry. That was for testing. I'm going to remove it now

Comment (Clebal) [reply packages/react-widgets/src/widgets/LegendWidget.js]: Updated.

**Comment by Clebal** [comment]
We need to discuss if it's interesting to have a initial* prop.

@padawannn @VictorVelarde @bbecquet

**Review by padawannn — APPROVED** [review]

**Pull request merged**